examples/mixed_hmm/model.py

```diff
@@ -0,0 +1,527 @@
+    """
+    Convert raw observation tensors into funsor.torch.Tensors
+    """
+    batch_inputs = OrderedDict([
+        ("i", bint(config["sizes"]["individual"])),
+        ("g", bint(config["sizes"]["group"])),
+        ("t", bint(config["sizes"]["timesteps"])),
+    ])
+
+    observations = {}
+    for name, data in config["observations"].items():
+        observations[name] = Tensor(data[..., :config["sizes"]["timesteps"]], batch_inputs)
+
+    return observations
+
+
+def initialize_zi_masks(config):
+    """
+    Convert raw zero-inflation masks into funsor.torch.Tensors
+    """
+    batch_inputs = OrderedDict([
+        ("i", bint(config["sizes"]["individual"])),
+        ("g", bint(config["sizes"]["group"])),
+        ("t", bint(config["sizes"]["timesteps"])),
+    ])
+
+    zi_masks = {}
+    for name, data in config["observations"].items():
+        mask = (data[..., :config["sizes"]["timesteps"]] == config["MISSING"]).to(
+            data.dtype)
+        zi_masks[name] = Tensor(1. - mask, batch_inputs)  # , bint(2))
+
+    return zi_masks
+
+
+def initialize_raggedness_masks(config):
+    """
+    Convert raw raggedness tensors into funsor.torch.Tensors
+    """
+    batch_inputs = OrderedDict([
+        ("i", bint(config["sizes"]["individual"])),
+        ("g", bint(config["sizes"]["group"])),
+        ("t", bint(config["sizes"]["timesteps"])),
+    ])
+
+    raggedness_masks = {}
+    for name in ("individual", "timestep"):
+        data = config[name]["mask"]
+        if len(data.shape) < len(batch_inputs):
+            while len(data.shape) < len(batch_inputs):
+                data = data.unsqueeze(-1)
+            data = data.expand(tuple(v.dtype for v in batch_inputs.values()))
+        data = data.to(config["observations"]["step"].dtype)
+        raggedness_masks[name] = Tensor(data[..., :config["sizes"]["timesteps"]],
+                                        batch_inputs)
+
+    return raggedness_masks
+
+
+def guide_sequential(config):
+    """generic mean-field guide for continuous random effects"""
+    params = initialize_guide_params(config)
+
+    N_c = config["sizes"]["group"]
+    N_s = config["sizes"]["individual"]
+
+    log_prob = Tensor(torch.tensor(0.), OrderedDict())
+
+    plate_g = Tensor(torch.zeros(N_c), OrderedDict([("g", bint(N_c))]))
+    plate_i = Tensor(torch.zeros(N_s), OrderedDict([("i", bint(N_s))]))
+
+    if config["group"]["random"] == "continuous":
+        with interpretation(eager):
+            eps_g_dist = plate_g + dist.Normal(**params["eps_g"])(value="eps_g")
+
+        log_prob += eps_g_dist
+
+    # individual-level random effects
+    if config["individual"]["random"] == "continuous":
+        with interpretation(eager):
+            eps_i_dist = plate_g + plate_i + dist.Normal(**params["eps_i"])(value="eps_i")
+
+        log_prob += eps_i_dist
+
+    return log_prob
+
+
+def model_sequential(config):
+    """
+    Simpler version of generic model with no zero-inflation
```

Comment:
note: if you do want to model zero-inflation, you could use `Stack`, e.g. as I've done in the BART model https://github.com/pyro-ppl/sandbox/blob/55ff040/2019-08-time-series/bart/forecast.py#L201
```py
likelihood = fdist.Categorical(gate, value="gated")
likelihood += funsor.Stack("gated", (
    fdist.Poisson(rate, value=data),
    fdist.Delta(0, value=data)))
likelihood = likelihood.reduce(ops.logaddexp, "gated")
```